pallets/crowdloans/src/lib.rs

```diff
@@ -258,6 +258,8 @@ pub mod pallet {
         ) -> DispatchResult {
             T::CreateVaultOrigin::ensure_origin(origin)?;
 
+            ensure!(cap > 0, Error::<T>::InsufficientBalance);
```

Comment:
we may need to give more explicit error. and `cap > 0` can be replaced by `!cap.is_zero()`